### ISSUE TITLE
Validation error with types differing in ArrayStride decoration

### ISSUE THREAD

**Issue by ShabbyX** [KhronosGroup/SPIRV-Tools#4273]
We encountered a SPIR-V validation error in ANGLE, and it's not very clear to me whether it's a glslang bug (i.e. bad code generated) or validation bug. Please see issuetracker.google.com/182568639 where you can find the SPIR-V that's failing validation (in comment 8) + my analysis (in comment 9).

The short of it is the following. The source shader has a uniform block containing a vec3[9]. This is given type %168 with decoration ArrayStride 16. There's a function that expects a vec3[9], which is given type %93. The function call site does:

```
%941 = OpLoad %168 ...
... = OpFunctionCall ... %941 ...
```

SPIR-V validation is complaining that:

    OpFunctionCall Argument <id> '941[%941]'s type does not match Function <id> '93[%_arr_v3float_uint_9]'s parameter type

In other words, it's passed a parameter of type %168, but expected type %93. Who's at fault here? Should glslang have generated different code (and if so, what?), or is SPIR-V validation wrong in declaring a mismatch between the two types?

**Comment by alan-baker** [comment]
This looks to me like a glslang bug. After the load of `%941`, glslang should copy it (either member-wise or using OpCopyLogical if the version is 1.4 or later) to an array with no stride decoration. The validator is correct that the types must match.

**Comment by ShabbyX** [comment]
Thank you Alan, I filed a bug with glslang.
